src/classes/ADDR_Addresses_TDTM.cls

```diff
@@ -1088,6 +1076,88 @@ public class ADDR_Addresses_TDTM extends TDTM_Runnable {
+
+        // Update the Address Mailing State with Billing/Mailing State Code accordingly
+        updateEmptyOrMismatchedAddressMailingStateWithAccountOrContactStateCode(sobjSrc, strFieldPrefixSrc, addr); 
+    }
+
+
+    /*******************************************************************************************************
+    * @description Updates the Address record's MailingCountry field when it is empty
+    *              with the Account's BillingCountryCode or Contact's MailingCountryCode value.
+    * @param sobjSrc the source Account or Contact
+    * @param strFieldPrefix the address fields to copy to, ie., Mailing, Other, Billing, Shipping
+    * @param addr the Address object to update accordingly
+    * @return void
+    */
+    @testVisible
+    private static void updateEmptyAddressMailingCountryWithAccountOrContactCountryCode(sObject sobjSrc, String strFieldPrefixSrc, Address__c addr) {
+        // Lookup Country if needed, by Country Code
+        String strCountryCode = (String)sobjSrc.get(strFieldPrefixSrc + 'CountryCode');
+
+        if (String.isNotBlank(strCountryCode) && String.isEmpty(addr.MailingCountry__c)) {
+            addr.MailingCountry__c = validCountriesByCode.get(strCountryCode.toUpperCase());
+        }
+    }
+
+
+    /*******************************************************************************************************
+    * @description Updates the Address record's MailingState fields when it is empty
+    *              and/or mismatched with the Account's BillingStateCode or Contact's MailingStateCode value.
+    * @param sobjSrc the source Account or Contact
+    * @param strFieldPrefix the address fields to copy to, ie., Mailing, Other, Billing, Shipping
+    * @param addr the Address object to update accordingly
+    * @return void
+    */
+    @testVisible
+    private static void updateEmptyOrMismatchedAddressMailingStateWithAccountOrContactStateCode(sObject sobjSrc, String strFieldPrefixSrc, Address__c addr) {
+        // Lookup State if needed, by Country Code and State Code.
+        String strStateCode = (String)sobjSrc.get(strFieldPrefixSrc + 'StateCode');
+
+        if (String.isNotBlank(strStateCode)
+            && (String.isEmpty(addr.MailingState__c) || !isMatchingStateCode(strStateCode, addr.MailingState__c))) {
+
+            // Assume it's a valid statecode
+            addr.MailingState__c = strStateCode;
+        }
+    }
+
+
+    /*******************************************************************************************************
+    * @description Determines if the Contact's or Account's state code matches the Addresses Mailing State
+    *              by looking up
+    * @param strStateCode the Contact's or Account's state code
+    * @param addr the Address object to update accordingly
+    * @return void
+    */
+    @testVisible
+    private static Boolean isMatchingStateCode(String strStateCode, String addressMailingState) {
```

Comment:
Suggestion: we might not use Hungarian Notation in the new methods, so maybe the `stateCode` can be as descriptive as `strStateCode` ?

Reply:
Yes, I can update the variable names.